### PR TITLE
Add schema compare generate script operation

### PR DESCRIPTION
**Pull request by kisantia**
This adds the schema compare generate script operation that generates the script for deploying the source to the target. It generates the script with the default options and does not yet support include/exclude differences. The public schema compare API only supports generating the script when the target is a database.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs [hunk 1]
```diff
@@ -85,6 +100,59 @@ await requestContext.SendResult(new SchemaCompareResult()
...
+                sqlTask = SqlTaskManagerInstance.CreateAndRun<SqlTask>(metadata);
+
+                await requestContext.SendResult(new ResultStatus()
+                {
+                    Success = true,
+                    ErrorMessage = operation.ErrorMessage
+                });
+            }
+            catch (Exception e)
+            {
```
Comment (udeeshagautam) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs]: We are not sending back e.Message. Might miss on some failure message..

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs [hunk 2]
```diff
@@ -73,9 +83,14 @@ await requestContext.SendResult(new SchemaCompareResult()
                             Differences = operation.Differences
```
Comment (udeeshagautam) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs]: Will operation.diff be available here in all cases (considering async/failure without exception if any)?

**Comment by udeeshagautam** [comment]
Looks like the cc decrease is in schema comapar area. Am I seeing it wrong? I do see tests so not sure if we need to do something there.

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs (hunk 1 above)
Comment (kisantia) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs]: operation.ErrorMessage gets set to e.Message before the exception gets thrown to this catch

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs (hunk 2 above)
Comment (kisantia) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareService.cs]: It should be ok if operation.Differences is null as long as it's handled properly in the ADS extension

**Pull request merged**